R/nm-join.R

```diff
@@ -265,3 +287,126 @@ can_be_nm_joined <- function(.mod){
     return(invisible(TRUE))
   }
 }
+
+
+#' Join simulation and input data
+#' @inheritParams setup_sim_run
+#' @param .join_col Character column name(s) to use to join table files.
+#'  Defaults to `NUM`. See Details.
+#' @param .cols_keep Either `'all'`, or a vector of column name(s) to retain
+#'  in the final dataset after joining. Defaults to keeping all columns.
+#' @inheritParams nm_file_multi_tab
+#'
+#' @note The join column name(s) specified should match what you provided to
+#' [new_sim_model()].
+#'
+#' @template nm-join-col
+#' @template nm-join-col-tables
+#' @template nm-join-col-eg
+#'
+#' @seealso [new_sim_model()], [nm_file_multi_tab()], [nm_tables()]
+#'
+#' @examples
+#' \dontrun{
+#'
+#' .sim_mod <- new_sim_model(.mod, .join_col = c("NUM", "ID"))
+#'
+#' submit_model(.sim_mod)
+#'
+#' nm_join_sim(.sim_mod, .join_col = c("NUM", "ID"), .cols_keep = "ID")
+#' }
+#'
+#'
+#' @return a tibble
+#' @export
+nm_join_sim <- function(
+    .mod,
+    .join_col = "NUM",
+    .cols_keep = "all",
+    add_table_names = FALSE
+){
+  checkmate::assert_string(.join_col)
+  checkmate::assert_string(.cols_keep)
+  checkmate::assert_logical(add_table_names)
+
+  # Only support joining the table we add so we can make certain assumptions
+  files <- nm_table_files(.mod)
+  if(length(files) > 1){
+    rlang::abort(
+      "`nm_join_sim` only supports joining a single table file (the one bbr adds)",
+      "Try using `nm_tables()` to manually join the tables"
+    )
+  }
+
+  # Support bbi_nonmem_models as well in the event the model was set up manually
+  check_model_object(.mod, c(NMSIM_MOD_CLASS, NM_MOD_CLASS))
+
+  # Get list of all tables. Likely only one simulation table, but may include
+  # other manually added tables
+  df_list <- nm_tables(
```

Comment:
I feel like we could write just a small amount of code that would read in our (single) `.tab` file from the simulation. It would be more robust and could utilize our knowledge about (1) the expected number of simulation replicates and (2) the size of the parent data set (potentially).